arm-software/linux/docs/PortingFromACfL.md

```diff
@@ -0,0 +1,101 @@
+|Base OpenMP Specification |OpenMP 4.0                       |OpenMP 2.5<br>Note: OpenMP support is experimental                                                   |
+|Parameterized Derived Type|Supported                        |PDT Kind - Supported<br>PDT Length - Not Supported                                                   |
+|Preprocessor              |Use -cpp to switch ON            |Always ON<br>Use `-cpp` to switch ON processing of predefined macros and macros from the command line|
+|Line length               |2100                             |Unlimited                                                                                            |
+|Recursive functions       |Use -frecursive                  |Default                                                                                              |
+|Main function             |Is a library linked at link-time |Is generated in the object file containing the program statement                                     |
+
+#### Difference in command line flags
+
+The following table summarises some of the most commonly used compiler flags in
+gfortran and gives their equivalent in the Arm Fortran compiler:
+
+|ACfL              |ATfL                      |Description                                                                                        |
+|------------------|--------------------------|---------------------------------------------------------------------------------------------------|
+|`-module <path>`  |`-J <path>`<br>`-I <path>`|Specifies a directory to place module files<br>Specifies a directory to search for module files    |
+|`-Mallocatable=03`|`-frealloc-lhs`           |Use Fortran 2003 standard semantics for assignments to allocatables                                |
+|`-Mallocatable=95`|`-fno-realloc-lhs`        |Use pre-Fortran 2003 standard semantics for assignments to allocatables                            |
+|`-r8`             |`-fdefault-real-8`        |Sets the default `KIND` for `REAL` and `COMPLEX` declarations, constants, functions, and intrinsics|
+|`-i8`             |`-fdefault-integer-8`     |Set the default `KIND` for `INTEGER` and `LOGICAL` to 64bit (i.e., `KIND = 8`)                     |
+
+#### Pre-defined macros
+
+`armflang` has the following compiler and machine specific predefined processor
+macros:
+
+|ACfL                  |ATfL                  |Value        |Description                                               |
+|----------------------|----------------------|-------------|----------------------------------------------------------|
+|`__FLANG`             |`__flang__`           |`1`          |Selection of compiler dependent source at compile time    |
+|`__arch64`            |N/A                   |`1`          |Selection of architecture dependent source at compile time|
+|`__aarch64__`         |N/A                   |`1`          |Selection of architecture dependent source at compile time|
```

Comment:
is there no way in LLVM flang to check which processor architecture is being used/to check for aarch64?

Reply:
good question. I've copy-pasted it from our Confluence pages.

Reply:
@kiranchandramohan do you know of existence of any such constant?